commands/proxy.ts

```diff
@@ -0,0 +1,137 @@
+import * as commandParams from "../command-params";
+import { isInteractive } from "../helpers";
+import { EOL } from "os";
+
+const proxyGetCommandName = "proxy|*get";
+const proxySetCommandName = "proxy|set";
+const proxyClearCommandName = "proxy|clear";
+
+export abstract class ProxyCommandBase implements ICommand {
+	public disableAnalytics = true;
+	public allowedParameters: ICommandParameter[] = [];
+
+	constructor(protected $analyticsService: IAnalyticsService,
+		protected $logger: ILogger,
+		protected $proxyService: IProxyService,
+		private commandName: string) {
+	}
+
+	public abstract execute(args: string[]): Promise<void>;
+
+	protected async tryTrackUsage() {
+		try {
+			await this.$analyticsService.trackFeature(this.commandName);
+		} catch (ex) {
+			this.$logger.trace("Error in trying to track proxy command usage:");
+			this.$logger.trace(ex);
+		}
+	}
+}
+
+export class ProxySetCommand extends ProxyCommandBase {
+	public allowedParameters = [new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector), new commandParams.StringCommandParameter(this.$injector)];
+
+	constructor(private $errors: IErrors,
+		private $injector: IInjector,
+		private $prompter: IPrompter,
+		protected $analyticsService: IAnalyticsService,
+		protected $logger: ILogger,
+		protected $proxyService: IProxyService) {
+		super($analyticsService, $logger, $proxyService, proxySetCommandName);
+	}
+
+	public async execute(args: string[]): Promise<void> {
+		let hostname = args[0],
+			port = args[1],
+			username = args[2],
+			password = args[3];
+
+		const noHostName = !hostname;
+		const noPort = !port;
+		const credentialsRequired = username && !password;
+
+		if (!isInteractive() && (noHostName || noPort || credentialsRequired)) {
+			this.$errors.fail("Console is not interactive - you need to supply all command parameters.");
```

Comment:
Do you think it would be nicer to the user if we pin-point to the missing pieces? What's our common practice? Or shall we at least display the syntax of the command `tns proxy hostname port username password`